src/Sentry/ExperimentalMetricsOptions.cs

```diff
@@ -0,0 +1,37 @@
+namespace Sentry;
+
+/// <summary>
+/// Settings for the experimental Metrics feature. This feature is preview only and will very likely change in the future
+/// without a major version bump... so use at your own risk.
+/// </summary>
+public class ExperimentalMetricsOptions
+{
+    /// <summary>
+    /// Determines whether code locations should be recorded for Metrics
+    /// </summary>
+    public bool EnableCodeLocations { get; set; } = true;
+
+    private IList<SubstringOrRegexPattern> _captureInstruments = new List<SubstringOrRegexPattern>();
+
+    /// <summary>
+    /// <para>
+    /// A customizable list of <see cref="SubstringOrRegexPattern"/>s defining which Instruments should be collected and
+    /// reported to Sentry.
+    /// </para>
+    /// <para>
+    /// These can be either custom System.Diagnostics.Metrics that you have instrumented in your
+    /// application or any of the built in metrics that are available.
+    /// </para>
+    /// <para>
+    /// See https://learn.microsoft.com/en-us/dotnet/core/diagnostics/built-in-metrics for more information.
+    /// </para>
+    /// </summary>
+    public IList<SubstringOrRegexPattern> CaptureInstruments
```

Comment:
is `Instruments` the best name? 

Also: Will we have some built-in ones here? And if so, we can't get code locations for those?

Reply:
I used Instruments since the `System.Diagnostics.Metrics` API uses [Instrument](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.instrument?view=net-8.0) as the base class for all System Diagnostic Metrics. 

Potentially `CaptureSystemDiagnosticsMetrics` would be more obvious/explicit?

In terms of defaults, we could capture none/some/all of [these build in metrics](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/built-in-metrics). 

Are there quotas that we need to be mindful of (so are we potentially going to chew through someone's entire Sentry Quota if we start collecting [routing match attempt](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/built-in-metrics-aspnetcore#metric-aspnetcoreroutingmatch_attempts) metrics?

If not, all of those built in metrics are potentially useful/interesting to people so we could enable these.

That raises another issue about the current API for enabling/disabling metrics. Currently this property is a bunch of SubstringOrRegexPatterns that match against the Instrument.Name. This is how you manually instrument stuff:
```
    private static readonly Meter HatsMeter = new(name: "HatCo.HatStore", "1.0.0");
    private static readonly Counter<int> HatsSold = HatsMeter.CreateCounter<int>(
        name: "hats-sold",
        unit: "Hats",
        description: "The number of hats sold in our store");
```

In the built in metrics you can see that related instruments often share the same prefix, so we could have a bunch of regex expresses to match those. 

Alternatively, those instruments were all created by the same meter, so we could also offer the ability to match against the meter name... that would make it easier for us to configure broad categories of built in metrics that we want to listen to (and implicitly easier for users to do the same) so sounds like a good idea.

I think it'd be an inclusive/or match (so we listen to the metric **either** if we match on the `Instrument` name **or** if we match on the `Meter` name). Possibly we'd want a separate options for each of those, in which case `CaptureSystemDiagnosticsMetrics` wouldn't be explicit enough... we'd probably want separate `CaptureSystemDiagnosticsInstruments` and `CaptureSystemDiagnosticsMeters` options. 

As I write this, I'm persuading myself that this is a good idea and those are probably good names 😉

Reply:
> Are there quotas that we need to be mindful of (so are we potentially going to chew through someone's entire Sentry Quota if we start collecting routing match attempt metrics?

Very good point, but we can have some built in metrics that don't count towards quota perhaps, if we don't have high cadinality, basically 0 tags.

> Potentially CaptureSystemDiagnosticsMetrics would be more obvious/explicit?

Not a bad idea, but maybe ur original one is better. dunno @bitsandfoxes thoughts?

> Possibly we'd want a separate options for each of those,

That makes sense. 

How about:

```
SystemDiagnosticsMetrics.Instruments = [".."];
SystemDiagnosticsMetrics.Meters = [".."];
```
Two properties and have `SystemDiagnosticsMetrics` be non nullable?